xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-ui/src/main/resources/AppWithinMinutes/LiveTableEditSheet.xml

```diff
@@ -150,8 +150,8 @@
     &lt;/dt&gt;
     &lt;dd&gt;#displayPropertyValue('description')&lt;/dd&gt;
     &lt;dt&gt;
-      &lt;label for="AppWithinMinutes.LiveTableClass_0_columns"&gt;$doc.displayPrettyName('columns', false, false)&lt;/label&gt;
-      &lt;span class="xHint"&gt;$services.localization.render('platform.appwithinminutes.liveTableEditorColumnsHint')&lt;/span&gt;
+      &lt;label for="availableColumns"&gt;$doc.displayPrettyName('columns', false, false)&lt;/label&gt;
```

Comment:
There is actually an input in the generated HTML that has the id `AppWithinMinutes.LiveTableClass_0_columns` and that is a much better fit for this label. Also, speaking about progressive enhancement (in the forum), this form actually uses progressive enhancement here, without JavaScript, this input is actually visible and the right target for this label. Only with JavaScript, the select that you target now is shown. So I think the correct solution here would be to change the `for`-attribute in JavaScript once the select has been made visible.

I'm also wondering if the select shouldn't get a different label as this select is just for adding new columns. There is another widget for arranging columns - making that widget accessible is probably a completely different challenge, though, as it cannot even be focused at the moment.

Reply:
> There is actually an input in the generated HTML that has the id AppWithinMinutes.LiveTableClass_0_columns and that is a much better fit for this label. Also, speaking about progressive enhancement (in the forum), this form actually uses progressive enhancement here, without JavaScript, this input is actually visible and the right target for this label. Only with JavaScript, the select that you target now is shown. So I think the correct solution here would be to change the for-attribute in JavaScript once the select has been made visible.

Addressed in https://github.com/xwiki/xwiki-platform/pull/2271/commits/377ca41ceb8269a9db8a857fbc03f1868024e4b5

> I'm also wondering if the select shouldn't get a different label as this select is just for adding new columns. There is another widget for arranging columns - making that widget accessible is probably a completely different challenge, though, as it cannot even be focused at the moment.

I think it's okay, because this select is actually just a select and the action link to validate the selection is properly labelled.
Labelled with "Add selected column".
![20988-labelIsEnough](https://github.com/xwiki/xwiki-platform/assets/28761965/7e27fd11-e8f3-4e97-afbb-86f35592cbdd)